IFDM-90: Added in a debt calculator

jenbreese opened the pull request:
# READY FOR REVIEW

# Summary
- Creating the debt calculator.
- features:
- - Tabbed views at the top
- - Info icon hover/click popovers

# Review By (Date)
- 11/21

# Criticality
- High

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. rebuild refresh
3. Verify it matches the design in ticket
4. Verify the math is right. Formulas are in the ticket
5. Double check the compounding since that has been problematic in the past

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

## Backend / Functional Validation
### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
[- IFDM-90](https://stanfordits.atlassian.net/browse/IFDM-90)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

jenbreese committed:
IFDM-90: Added in a debt calculator

jenbreese committed:
worked on layout of the page.

jenbreese committed:
fixup

jenbreese committed:
IFDM-90: additional styles

jenbreese committed:
fixup

jenbreese committed:
fixes to color, font sizes and ui

jenbreese committed:
fixup to slider dark mode color

jenbreese committed:
adding in popover funcationality.

rebeccahongsf reviewed (commented):
The calculations appear to be correct! 🎉 

I did notice a few things on smaller screen sizes and had some questions/feedback 🤔 

1. Can the tool tips be fixed to be responsive?The tool tips bleed outside of the screen view:
<img width="563" height="414" alt="image" src="https://github.com/user-attachments/assets/a74df65d-cb10-494c-a20b-8452b46d31ff" />

2. I'm unsure if this is an a11y concern: The tool tips appear on hover but doesn't stay open on click, it's more noticeable for smaller screens when I use the browser inspect tool. The keyboard focus works great though!

3. For Calculate Required Payment, the "Total: 10 years and 0 months" in blue text under `Target time to payoff` is missing. 
Figma:
<img width="415" height="393" alt="image" src="https://github.com/user-attachments/assets/96fed8e1-f1f2-44c0-8ce2-80a325cfc5e2" />

Current site:
<img width="478" height="375" alt="image" src="https://github.com/user-attachments/assets/b9959f78-de3a-48e6-8737-86d1525e7471" />

3. Are the buttons suppose to stack or appear side-by-side on smaller screens? 
Figma:
<img width="389" height="333" alt="image" src="https://github.com/user-attachments/assets/79b55a27-4734-40a3-ad22-37543a368ff6" />


Current site:
<img width="487" height="326" alt="image" src="https://github.com/user-attachments/assets/17941665-f033-4a69-87b6-d462fa1d5aba" />

4. There are some spacing changes that differ from the figma mock like “Your Payoff Summary” and “Required Payment” appear to extend edge-to-edge on smaller screens. I’m not sure how strict the client is about spacing but figured I'd call it out (personally, I prefer keeping some margin vs. going full edge-to-edge 😬)
Figma:
<img width="1366" height="1298" alt="image" src="https://github.com/user-attachments/assets/460864da-24d9-4300-8519-d5e68af5b819" />

jenbreese committed:
fixup from feedback

jenbreese commented:
Thank you @rebeccahongsf! 
1. There aren't designs for what these popovers look like so I moved it to the top/left. I also changed the styling a bit. 
2. I will look at.
3. I fixed.
3. I stacked the buttons because there are so many words i thought it was nicer than side by side.
4. The client is pretty open to feedback on the design.  This one matches how we do the other calculators. 

I'll look at the click/hover problem and hopefully merge this sometime tomorrow so the client can look and give feedback on the design.  We often iterate that way.

jenbreese committed:
fix to a click and no hover. fix to colors for dark mode

rebeccahongsf reviewed (approved):
Thank you for the response and changes, @jenbreese! 😄 Sounds good on all points. I'll go ahead and approve so you can get this out to the client.
